SolrNet.Cloud.Tests/FakeOperations.cs

```diff
@@ -464,6 +494,31 @@ public Task<ResponseHeader> DeleteAsync(IEnumerable<string> ids, DeleteParameter
         }
 
         public Task<ResponseHeader> DeleteAsync(IEnumerable<string> ids, ISolrQuery q)
+        {
+            SetLastOperation(); return null;
+        }
+
+        public Task<ResponseHeader> AtomicUpdateAsync(string uniqueKey, string id, IEnumerable<AtomicUpdateSpec> updateSpecs, AtomicUpdateParameters parameters)
+        {
+            SetLastOperation(); return null;
+        }
+
+        public Task<ResponseHeader> AtomicUpdateAsync(T doc, IEnumerable<AtomicUpdateSpec> updateSpecs)
+        {
+            SetLastOperation(); return null;
+        }
+
+        public Task<ResponseHeader> AtomicUpdateAsync(string id, IEnumerable<AtomicUpdateSpec> updateSpecs)
+        {
+            SetLastOperation(); return null;
+        }
+
+        public Task<ResponseHeader> AtomicUpdateAsync(T doc, IEnumerable<AtomicUpdateSpec> updateSpecs, AtomicUpdateParameters parameters)
+        {
+            SetLastOperation(); return null;
+        }

```

Comment:
Returning `null` in an `Async` method is not the same as returning `null` in a regular method. Because without the `async` keyword in the method, you are responsible for returning the `Task` that can be awaited. In this code, with the `async` keyword missing, the `Task` that is returned is `null`, so the `await` will blow up.

In order for this to work, either add the async keyword as in:
``` csharp
public async Task<ResponseHeader> AtomicUpdateAsync(string id, IEnumerable<AtomicUpdateSpec> updateSpecs)
{
     SetLastOperation(); return null;
}
```
or return a valid `Task`
``` csharp
public Task<ResponseHeader> AtomicUpdateAsync(string id, IEnumerable<AtomicUpdateSpec> updateSpecs)
{
     SetLastOperation(); return Task.FromResult<ResponseHeader>(null);
}
```

Reply:
Hi gjunge, sorry for slow reply.

Good spot! I'm assuming this also applies to the other async tests in that class. I will fix them all and submit an update to the atomic-updates feature branch.

Reply:
Pull request submitted.
https://github.com/SolrNet/SolrNet/pull/351